File: C++/050. Pow(x, n).cpp
class Solution {
public:
    double myPow(double x, int n) 
    {
        if (n == 0) return 1;
        else if (n == 1) return x;
        else if (n == -1 && x != 0) return 1/x;
        double temp1 = myPow(x, n>>1);
        double temp2 = myPow(x, n&1);
        return temp1 * temp1 * temp2;
    }
};

File: python/050. Pow(x, n).py
class Solution(object):
    def _myPow(self, x, n):
        if n == 1: return x
        val1 = self._myPow(x, n/2)
        val2 = x if n & 1 else 1
        return val1 * val1 * val2
    
    def myPow(self, x, n):
        """
        :type x: float
        :type n: int
        :rtype: float
        """
        if n == 0: return 1
        if x == 0: return 0
        if n < 0: x = 1.0 / x
        return self._myPow(x, abs(n))